### PR TITLE
kmeans (feature): split select function by blocks in external loop

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
This splitting is necessary to support huge datasets in KMeans

### CODE REVIEW

**Comment by Alexandr-Solovev** [comment]
It is second option of the https://github.com/oneapi-src/oneDAL/pull/2346 fix

File: cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp_impl.hpp [hunk 1]
```diff
@@ -150,56 +150,71 @@ sycl::event kernels_fp<Float>::select(sycl::queue& queue,
...
-                const std::int64_t sg_global_id = wg_id * sg_num + sg_id;
-                if (sg_global_id >= row_count)
-                    return;
-                const std::int64_t in_offset = sg_global_id * stride;
-                const std::int64_t out_offset = sg_global_id;
+    const std::int64_t num_rows =
+        std::min(static_cast<std::int64_t>(bk::device_max_mem_alloc_size(queue) / sizeof(Float)),
+                 static_cast<std::int64_t>(std::numeric_limits<int>::max() / wg_size));
+    std::vector<sycl::event> events;
+    for (std::int64_t start_row = 0; start_row < row_count; start_row += num_rows) {
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp_impl.hpp:157]:
```suggestion
const auto block_size = propose_block_size<Float>(queue, row_count);
const bk::uniform_blocking blocking(row_count, block_size);

std::vector<sycl::event> events(blocking.get_block_count());
for (std::int64_t b = 0; b < blocking.get_block_count(); ++b) {

    const auto first_row = blocking.get_block_start_index(b);
    const auto last_row = blocking.get_block_end_index(b);
```

Comment (KulikovNikita) [reply cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp_impl.hpp:157]: Please use well established functionality. Also I would recommend not to create so many events

**Review by KulikovNikita — APPROVED** [review]

**Review by ethanglaser — APPROVED** [review]

**Pull request merged**